branding: Updating branding page and Web theme

cwhite911 opened the pull request:
I'm updating the branding and web theme pages to demo the new logos/colors based on the discussion at the last PSC meeting.

![Screenshot 2025-03-25 at 16-04-36 Brand](https://github.com/user-attachments/assets/74282d33-9840-439e-8828-eace9e921806)

wenzeslaus commented on static/images/logos/gs-white.svg:
Can you please test behavior of this file in Inkscape? I get somewhat erratic behavior. The graphic is semi-transparent while reporting 100% opacity. There are no layers, when I create one and move the graphics there, it becomes white, but it also becomes bigger and the bottom becomes broken.

![Screenshot from 2025-03-12 22-41-05](https://github.com/user-attachments/assets/af0d5b58-d85d-4ff9-9333-6a25d5b83073)
![Screenshot from 2025-03-12 22-41-11](https://github.com/user-attachments/assets/4119daf9-83eb-45c7-a545-f65592656c45)

There is a mask set, if I release mask (with an unmodified file), I get:

![image](https://github.com/user-attachments/assets/a4c351b9-6675-438b-b6f9-8a2f7fa04821)

wenzeslaus commented on content/about/brand.md:
```diff
@@ -9,23 +9,22 @@ layout: "brand"
 <div class="row mt-30">
 <div class="col-lg-6 text-center panel">
-<img src="../../images/logos/grasslogo.svg" width="33%" alt="GRASS GIS">
+<img src="../../images/logos/gs-green.svg" width="33%" alt="GRASS">
```
Let's call the files grass... instead of gs...

cwhite911 replied:
I've generally been using gs for the GRASS assets namespace, logos, css variables. Is this just a personal preference or do you have another reason you want to rename it?

wenzeslaus replied:
These files will be downloaded by people, so they end up with "gs something" files possibly without context. Also, grass is not a long word for a filename. CSS property prefix or Python import alias are different because they have limited scope and are repeated often (I'm not sure about the other uses of gs). I do prefer explicit rather than implicit in naming.

cwhite911 replied:
Good point. I'll update them.

cwhite911 commented:
I've a script to take screenshots of the entire site. I've not added them to GitHub, but you can follow the docs here to generated them locally.

# Screenshots

To take screenshots of the site. Install node.js and puppeteer.

If you don't have node.js installed, I recommend using [nvm](https://github.com/nvm-sh/nvm) to install it. You can install nvm with the following command:

> I've only tested this with node v22.14.0 and npm v10.9.2

```bash
curl -o- https://raw.githubusercontent.com/nvm-sh/nvm/v0.40.2/install.sh | bash
```

After installing nvm, you can install node.js with the following command:

```bash
nvm use  # uses node version from .nvmrc
```

Once you have node installed, you can install puppeteer and other dependencies with the following commands:

```bash
npm install
```

Run hugo server to start the site:

```bash
hugo server
```

To take a screenshot of the site, run the following command:

```bash
npm run screenshots
```

The screenshots will be saved in the `screenshots/images`.

wenzeslaus commented on content/about/brand.md:
```diff
@@ -4,118 +4,38 @@ date: 2018-12-29T11:02:05+06:00
+    {{< brandColor colorName="White"     hex="#f7f7f7" >}}
+{{</ brandColorList >}}
+
+These are the **main colors** used for the  **GRASS logo**.
+The GRASS weed is green, frame and text are black.
+Monochrome version of the logo is also allowed.
```
```suggestion
```

wenzeslaus commented on content/about/brand.md:
```diff
@@ -4,118 +4,38 @@ date: 2018-12-29T11:02:05+06:00
-	</div>
-</div>
-
+Welcome to the GRASS branding guide! Here you can learn about how to use GRASS logos and colors.
```
```suggestion
Whether you are contributing to GRASS or creating materials which involve GRASS, help the project by keeping these branding guidelines.
```

wenzeslaus commented on content/about/brand.md:
```diff
@@ -4,118 +4,38 @@ date: 2018-12-29T11:02:05+06:00
+    {{< brandLogo colorName="White"        logo="/images/logos/grass-white.svg" >}}
+        The white logo should be used on green backgrounds, photographs of grass, and other pictures as long as they provide sufficient contrast. Use with the brand green color or a grass photograph in the background is preferred to maintain brand consistency.
+    {{</ brandLogo >}}
+    {{< brandLogo colorName="Blue"         logo="/images/logos/grass-blue.svg"  >}}
+        The secondary logo of the GRASS project is the blue logo. It is used in contexts where a different color scheme is needed, such as on darker backgrounds or in materials where the primary green logo does not fit the design aesthetics.
+    {{</ brandLogo >}}
```
Remove the blue logo. I didn't come up with any use for it, so that's why I didn't include it in the draft.

```suggestion
```

cwhite911 commented:
@wenzeslaus here is the latest iteration of the branding page.

![screenshot_93_aboutbrand](https://github.com/user-attachments/assets/f1b45481-3c97-44e9-af8f-a9f4b4238a40)

wenzeslaus commented:
This looks good. We still need the horizontal variants and variants without text. Short description should be already in the working document.

Please, for the final screenshot of the branding, crop it to include only the Brand page itself with Logos and Colors (but not the Web theme).

cwhite911 commented:
@wenzeslaus 

![Screenshot 2025-03-25 at 16-04-36 Brand](https://github.com/user-attachments/assets/74282d33-9840-439e-8828-eace9e921806)

wenzeslaus reviewed (dismissed):
The old favicon picture is still visible in the page header. You should delete that file completely. It was never an official logo variant, so no need to keep it around.

hellik commented:
> @wenzeslaus
> 
> ![Screenshot 2025-03-25 at 16-04-36 Brand](https://private-user-images.githubusercontent.com/3803800/426716993-74282d33-9840-439e-8828-eace9e921806.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3NDMwMTk1NDksIm5iZiI6MTc0MzAxOTI0OSwicGF0aCI6Ii8zODAzODAwLzQyNjcxNjk5My03NDI4MmQzMy05ODQwLTQzOWUtODgyOC1lYWNlOWU5MjE4MDYucG5nP1gtQW16LUFsZ29yaXRobT1BV1M0LUhNQUMtU0hBMjU2JlgtQW16LUNyZWRlbnRpYWw9QUtJQVZDT0RZTFNBNTNQUUs0WkElMkYyMDI1MDMyNiUyRnVzLWVhc3QtMSUyRnMzJTJGYXdzNF9yZXF1ZXN0JlgtQW16LURhdGU9MjAyNTAzMjZUMjAwMDQ5WiZYLUFtei1FeHBpcmVzPTMwMCZYLUFtei1TaWduYXR1cmU9NTliYzA1YTYwMDdkOWE3MGY5MDRkNmY3YzMzNWQwYWNmZDE2ZDg1ZjhhZGQwNjA2YmFmZWZlZWU4YjA3MDBiZCZYLUFtei1TaWduZWRIZWFkZXJzPWhvc3QifQ.G-KmHojrZBG7QuTjts2RkgBd9O_nnRp9q-FTujX0t-Y)

looks good! :-) thanks for the effort! also the gradien version works.

nilason commented:
Great job, looks really nice!

(Minor nitpicking note: there are a number of files which do not end with newline, it would be good to correct that).

veroandreo commented:
All in all, I like it :)

Some observations:
- There's an "Old" in the "Join the GRASS community" picture in the carousel.
- Locally, the SVG logos in the brand page look funny sometimes... Do we really need white and light grey background versions? The page looks too crowded IMHO.  
- If we keep the plain green background in the main page behind the new logo, I'd remove the banner then... or use a different color for it, i.e., the banner. Or, can't we use the modified grass picture as background? How does that look like?

wenzeslaus commented on package.json:
```diff
@@ -0,0 +1,12 @@
+{
+  "name": "puppeteer-screenshots",
+  "type": "module",
+  "scripts": {
+    "screenshots": "node screenshots/screenshots.js"
+  },
+  "dependencies": {
+    "puppeteer": "^24.4.0",
+    "puppeteer-core": "^24.4.0",
+    "xml2js": "^0.6.2"
+  }
+}
```
Let's remove the screenshot tooling from this PR to make it smaller.

petrasovaa replied:
This is still there.

cwhite911 commented:
> Locally, the SVG logos in the brand page look funny sometimes...

@veroandreo Can you share a screenshot?

> There's an "Old" in the "Join the GRASS community" picture in the carousel.

I updated the carousel to the GRASS Developer Summit 2025.

## Update carosel
![image](https://github.com/user-attachments/assets/4c145c92-734b-4c87-94f7-a5b8accb5e11)

> f we keep the plain green background in the main page behind the new logo, I'd remove the banner then... or use a different color for it, i.e., the banner. Or, can't we use the modified grass picture as background? How does that look like?

## Index banner update
![image](https://github.com/user-attachments/assets/03270a0b-e777-4feb-b062-65d6d7f1ab80)

> (Minor nitpicking note: there are a number of files which do not end with newline, it would be good to correct that).
@nilason I've not address this yet but will before we merge.

##Brand

> Do we really need white and light grey background versions? The page looks too crowded IMHO.

Idk. Good thing it is easy to change so we can play with it some. I'll send a version without.

![image](https://github.com/user-attachments/assets/1eef233f-f529-430b-8c6c-a1cfef55fb80)

veroandreo commented:
Thanks @cwhite911!! Home looks much better!! And I really like the picture of the summit in the carousel!

3 suggestions and I shut up :face_in_clouds:, i.e., after addressed I think this is ready to be merged :star_struck:  
- The summit pic appears to change the size of the carousel box compared to the other pics in it
-  I'd use the plain green logo here: 
![image](https://github.com/user-attachments/assets/28a6de9f-1ed4-4306-9722-49e4fd939e6b)
- I'd remove the grey background versions from brand, IMHO, those are not needed, they occupy space making the page longer, they are barely distinguishable and since they are svg files, people can just download the SVG and change background themselves, no? 

The SVG logos look fine now, not funny effect anymore

lindakarlovska commented:
> > Nice work! Overall, I like it. In which situations colors Grey, Alt-Grey, Dark Grey and Black are supposed to be used? (I think they are not part of any of the logos.)
> 
> @lindakarlovska They are used in the overall design, but mostly for different shading options and text. In terms of the "brand" we have had shirts with grey/black fabric and white logos before. We can also hide the grey colors from this page and use just use them internally.

Yep @cwhite911, having in mind @veroandreo's improver of getting rid of grey backgrounds in logos, I would suggest hiding the whole Grey color row. It corresponds to the logos above and the page gonna be shorter. Otherwise, I like the whole new design, including the carosel. :-)

petrasovaa commented on themes/grass/assets/css/style.css:
```diff
@@ -1,62 +1,177 @@
+  --gs-grey-light-color--lightest: rgba(240, 240, 241, 0.071);
+
+  /* Dark Gray */
+  --gs-dark-gray-color: rgb(117, 116, 115);
+  --gs-dark-gray-color--light: rgba(117, 116, 115, 0.541);
+  --gs-dark-gray-color--lighter: rgba(117, 116, 115, 0.322);
+  --gs-dark-gray-color--lightest: rgba(117, 116, 115, 0.071);
```
```suggestion
  --gs-grey-dark-color: rgb(117, 116, 115);
  --gs-grey-dark-color--light: rgba(117, 116, 115, 0.541);
  --gs-grey-dark-color--lighter: rgba(117, 116, 115, 0.322);
  --gs-grey-dark-color--lightest: rgba(117, 116, 115, 0.071);
```

petrasovaa reviewed (changes requested):
I would change the text to dark grey, I think the pitch black doesn't look very good. I was experimenting with it and it seems the dark grey there is not dark enough. Previously, it used #636363, which is darker than the current dark grey. The black-color-light also seems somewhat too light.  Was there a specific reason for the black?

cwhite911 commented:
@veroandreo and @lindakarlovska I've addressed the issues you pointed out.

Here are the changes on index.html

- [x] changed navigation logo color
- [x] Fixed issue with carousel image size

![image](https://github.com/user-attachments/assets/6db13cb2-16ef-46da-bdd1-7e79b1c4d4a6)

And the issues from the brand page.

- [x] Removed grey colors
- [x] Removed grey background images (Note: I tried keeping the horizontal logo to the right, but the spacing looked awkward)
- [x] Change color name to `-gs-grey-dark-color`

![image](https://github.com/user-attachments/assets/2eae110e-c4ea-4c81-be8a-48b20cf653ec)

cwhite911 commented:
Here is a screenshot from the last set of commits where I synced the `--gs-grey-dark-color` and the fonts with MkDocs.

![image](https://github.com/user-attachments/assets/e24a0fa3-f859-4f27-a7e3-fb010d61f433)

petrasovaa commented:
@cwhite911 do you want to merge this now or wait for anything else?

cwhite911 commented:
> @cwhite911 do you want to merge this now or wait for anything else?

I just updated some references to GRASS GIS on the home page. I think it is fine to merge. The other option is to wait until the docs synchronization and tutorial branding is complete so we can release the updates all at the same time.

wenzeslaus commented:
Let's merge this...and also prepare a news post, no?

cwhite911 commented:
> Let's merge this...and also prepare a news post, no?

Yes, we should prepare a news post and have some social post for the new branding/docs.

veroandreo commented:
I'm merging this one, ok? Great work @cwhite911 and all!
